pkg/crds/crds.go

```diff
@@ -26,142 +28,120 @@ import (
+	r := &CRDReconciler{
 		crds:   crds,
 		config: cfg,
-	}, nil
+	}
+
+	return r, nil
 }
 
-func InitCrds() []*extensionsobj.CustomResourceDefinition {
-	return []*extensionsobj.CustomResourceDefinition{
-		crdL("VirtualService", "VirtualServices", crdConfigs[Networking], "istio-pilot", "", "", extensionsobj.NamespaceScoped, true),
-		crdL("DestinationRule", "DestinationRules", crdConfigs[Networking], "istio-pilot", "", "", extensionsobj.NamespaceScoped, true),
-		crdL("ServiceEntry", "ServiceEntries", crdConfigs[Networking], "istio-pilot", "", "", extensionsobj.NamespaceScoped, true),
-		crdL("ClusterRbacConfig", "ClusterRbacConfigs", crdConfigs[Rbac], "istio-pilot", "rbac-istio-pilot", "rbac", extensionsobj.ClusterScoped, true),
-		crd("Gateway", "Gateways", crdConfigs[Networking], "istio-pilot", "", "", extensionsobj.NamespaceScoped),
-		crd("EnvoyFilter", "EnvoyFilters", crdConfigs[Networking], "istio-pilot", "", "", extensionsobj.NamespaceScoped),
-		crd("Policy", "Policies", crdConfigs[Authentication], "", "", "", extensionsobj.NamespaceScoped),
-		MeshPolicy(),
-		crd("HTTPAPISpecBinding", "HTTPAPISpecBindings", crdConfigs[Apim], "", "", "", extensionsobj.NamespaceScoped),
-		crd("HTTPAPISpec", "HTTPAPISpecs", crdConfigs[Apim], "", "", "", extensionsobj.NamespaceScoped),
-		crd("QuotaSpecBinding", "QuotaSpecBindings", crdConfigs[Apim], "", "", "", extensionsobj.NamespaceScoped),
-		crd("QuotaSpec", "QuotaSpecs", crdConfigs[Apim], "", "", "", extensionsobj.NamespaceScoped),
-		crd("rule", "rules", crdConfigs[Policy], "mixer", "istio.io.mixer", "core", extensionsobj.NamespaceScoped),
-		crd("attributemanifest", "attributemanifests", crdConfigs[Policy], "mixer", "istio.io.mixer", "core", extensionsobj.NamespaceScoped),
-		crd("RbacConfig", "RbacConfigs", crdConfigs[Rbac], "mixer", "istio.io.mixer", "rbac", extensionsobj.NamespaceScoped),
-		crd("ServiceRole", "ServiceRoles", crdConfigs[Rbac], "mixer", "istio.io.mixer", "rbac", extensionsobj.NamespaceScoped),
-		crd("ServiceRoleBinding", "ServiceRoleBindings", crdConfigs[Rbac], "mixer", "istio.io.mixer", "rbac", extensionsobj.NamespaceScoped),
-		crd("adapter", "adapters", crdConfigs[Policy], "mixer", "istio.io.mixer", "rbac", extensionsobj.NamespaceScoped),
-		crd("instance", "instances", crdConfigs[Policy], "mixer", "instance", "mixer-instance", extensionsobj.NamespaceScoped),
-		crd("template", "templates", crdConfigs[Policy], "mixer", "template", "mixer-template", extensionsobj.NamespaceScoped),
-		crd("handler", "handlers", crdConfigs[Policy], "mixer", "handler", "mixer-handler", extensionsobj.NamespaceScoped),
-		crd("Sidecar", "Sidecars", crdConfigs[Networking], "istio-pilot", "", "", extensionsobj.NamespaceScoped),
-		crd("AuthorizationPolicy", "authorizationpolicies", crdConfigs[Security], "isito-pilot", "", "security", extensionsobj.NamespaceScoped),
+func (r *CRDReconciler) LoadCRDs() error {
+	for _, file := range []string{"crd-10.yaml", "crd-11.yaml", "crd-14.yaml"} {
```

Comment:
Alternatively file names in the repo could be listed here, instead of hard-coding them. Later if new files are added they would not be forgotten in that case.

Reply:
👍

Reply:
Great, thanks!